frontend/src/utils/authorizedAxios.ts

```diff
@@ -29,30 +30,31 @@ authorizedAxios.interceptors.request.use(
 
 authorizedAxios.interceptors.response.use(
   (response) => response,
-  async function (error) {
-    const originalRequest = error.config;
-    // If token expires, replay request
-    if (error.response.status === 401 && !originalRequest._retry) {
-      originalRequest._retry = true;
-
-      const refreshToken = localStorage.getItem(REFRESH_TOKEN);
-      if (!refreshToken) {
-        return Promise.reject(error);
+  (error) =>
+    mutex.runExclusive(async () => {
+      const originalRequest = error.config;
+      // If token expires, replay request
+      if (error.response.status === 401 && !originalRequest._retry) {
+        originalRequest._retry = true;
+
+        const refreshToken = localStorage.getItem(REFRESH_TOKEN);
+        if (!refreshToken) {
+          throw error;
+        }
+
+        try {
+          const res = await refreshAccessToken(refreshToken);
+          localStorage.setItem(ACCESS_TOKEN, res.data.accessToken);
+          localStorage.setItem(REFRESH_TOKEN, res.data.refreshToken);
+        } catch {
+          throw error;
+        }
```

Comment:
catch should log the user out of the app. rn it redirects to onboarding page again if refresh token is also expired

Reply:
logout is handled by useCheckAuth now: authProvider.checkAuth checks validity of both tokens -> if fail -> useCheckAuth logs the user out